### ISSUE TITLE
express integration

### ISSUE THREAD

**Issue by mpareja** [mozilla/node-client-sessions#101]
The out-of-the-box experience for working with `client-sessions` on express behind an HTTP proxy is not great.

The requirement for `req.connection.proxySecure` doesn't seem to work well out-of-the-box with express.  Express has configuration around how to deal with proxied requests (the "trust proxy" settings). This express functionality does not set `req.connection.proxySecure`. It does however set `req.secure` to true for properly proxied requests.

Is this something you would consider adding support for, or should I consider writing separate middleware that bridges between `req.secure` and `req.connection.proxySecure` when "trust proxy" is enabled?

**Comment by ponelat** [comment]
@mpareja how did you solve this? I'm keen to know if you just wrote something like...
```js
if(req.secure) {
  req.connection.proxySecure = true
}
```

**Comment by mpareja** [comment]
@ponelat Sorry for getting back to you so late, your comment was lost in my inbox.

Yes, I added a middleware that does just that in tandem with setting `app.set('trust proxy', true);`.

**Comment by q42jaap** [comment]
@mpareja @ponelat 
I think I've set this up by specifying:
```javascript
clientSessions({
  ...
  cookie: {
    httpOnly: true,
    secureProxy: true
  }
}
```

**Comment by ponelat** [comment]
Yeah, thanks @mpareja and @q42jaap . I used the `app.set('trust proxy'. true)`

**Comment by mpareja** [comment]
@q42jaap @ponelat  While there is a README file saying `secureProxy`, I don't see any other references to that in this repo. We actually use the `secure` property:

```
clientSessions({
  ...
  cookie: {
    httpOnly: true,
    secure: true
  }
}
```

**Comment by patrick-m-m** [comment]
I was looking into this as well, before finding this issue. I discovered this in the underlying [Cookies module](https://www.npmjs.com/package/cookies) that client-sessions requires:

```
Cookies.prototype.set = function(name, value, opts) {
// snip
  if (opts && "secureProxy" in opts) {
    deprecate('"secureProxy" option; use "secure" option, provide "secure" to constructor if needed')
    cookie.secure = opts.secureProxy
  }
```
NPM resolved version 0.7.1 for the Cookies module when installing client-sessions (whose package.json specifies `^0.7.0`). So maybe at one point in time the `secureProxy` setting did what the README implies, but clearly no longer.

Thanks for the tip on the workaround middleware.
